include/boost/dll/detail/mem_fn_cast.hpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * mem_fn_cast.hpp
+ *
+ *  Created on: 17.10.2015
+ *      Author: Klemens
+ */
+
+#ifndef INCLUDE_BOOST_DLL_DETAIL_MEM_FN_CAST_HPP_
+#define INCLUDE_BOOST_DLL_DETAIL_MEM_FN_CAST_HPP_
+
+#include <boost/config.hpp>
+#include <boost/dll/detail/aggressive_ptr_cast.hpp>
+
+namespace boost
+{
+namespace dll
+{
+
+namespace detail
+{
+#if defined(BOOST_MSVC) || defined(BOOST_MSVC_VER)
+template<typename T> T mem_fn_cast(void* p)
+{
+	union
+	{
+		T out;
+		void* in;
+	} u;
+	u.in = p;
+	return u.out; //aggressive_ptr_cast<T>(p);
```

Comment:
Oh, good to know. I actually had an aggressive_ptr_cast in there, but that gave an error.

Just out of curiosity: why does that yield undefined behaviour? Can a union pad the members?